src/lxc/conf.c

```diff
@@ -1849,9 +1865,13 @@ static inline int mount_entry_on_generic(struct mntent *mntent,
 		return -1;
 	}
 
+	/* rootfs struct will be empty when container is created without rootfs. */
```

Comment:
For future reference: The only place where `rootfs` can legitimately be `NULL` is in `mount_entry_on_generic()` when `NULL` is passed from `mount_entry_on_systemfs()`. So I think we can simplify all of this significantly.